src/PayU/Alu/Request.php

```diff
@@ -58,29 +60,125 @@ class Request
     private $fx;
 
     /**
-     * @var array
+     * @var string
      */
-    private $internalArray;
+    private $paymentsApiVersion;
+
+    /**
+     * @var string
+     */
+    private $customUrl;
 
     /**
      * @param MerchantConfig $merchantConfig
      * @param Order $order
      * @param Billing $billing
      * @param AbstractCommonAddress $delivery
      * @param User $user
+     * @param string $paymentsApiVersion
      */
     public function __construct(
         MerchantConfig $merchantConfig,
         Order $order,
         Billing $billing,
         AbstractCommonAddress $delivery = null,
-        User $user = null
+        User $user = null,
+        $paymentsApiVersion = AluV3::API_VERSION_V3
     ) {
         $this->merchantConfig = $merchantConfig;
         $this->order = $order;
         $this->billingData = $billing;
         $this->deliveryData = $delivery;
         $this->user = $user;
+        $this->paymentsApiVersion = $paymentsApiVersion;
```

Comment:
By looking at how the Request object is being used, I think paymentsApiVersion would stay better somewhere else but not on this class. The request is something that should go into an AuthorizationInterface and be used there. But it not only does that, it is also used to create such an AuthorizationInterface. Furthermore, paymentsApiVersion is not needed anymore. By how it is set now it means that a Request with paymentsApiVersion  3 or 4 is a request specific to that version. But it is not true. The request is something generic and independent of the API version.  Maybe we could send the version when calling \PayU\Alu\Client::pay.

Reply:
\PayU\Alu\Client::pay being a method from the old namespace, here may be changes made to it  - params added in the signature for example and if we change it, may lead to code incompatibilities. I would leave it as it is right now.